core/src/main/java/smile/classification/SVM.java

```diff
@@ -1533,15 +1698,19 @@ private double posterior(LASVM svm, double y) {
         final double minProb = 1e-7;
         final double maxProb = 1 - minProb;
 
-        return Math.min(Math.max(svm.platt.predict(y), minProb), maxProb);
+        return Math.min(Math.max(scalingMethod == ScalingMethod.Platt? svm.platt.predict(y) : svm.isotonicRegression.predict(y), minProb), maxProb);
```

Comment:
no sure about this. simple prediction without postiori probability should not bother this at all.

Reply:
this is just a switch added on top of existing logic. If simple prediction without postiori probability does not need concern about scaling, why do we need existing platt scaling in this method?